src/abstractdataframe/abstractdataframe.jl

```diff
@@ -2531,33 +2531,95 @@ function _permutation_helper!(fun::Union{typeof(Base.permute!!), typeof(Base.inv
         if haskey(seen_cols, col)
             push!(toskip, i)
         else
+            checkbounds(col, eachindex(p))
```

Comment:
why do you think it is needed? This part of code can assume that each column in a data frame is 1-based and has equal length. We check that this invariant is met elsewhere. We might just check if length of `p` equals number of rows in `df`.

Reply:
Thanks! I did not know that we can assume the dataframe is one-indexed and every column has the same length. I removed that check and replaced it with a check that `length(p) <= nrow(df)` because short permutations are supported.

EDIT: moved from main thread